src/libraries/Microsoft.Extensions.Logging/gen/TemplateHelper.cs

```diff
@@ -0,0 +1,346 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Text
+{
+    internal static class TemplateHelper
+    {
+        /// <summary>
+        /// Parses the message template and throws exception for malformed messages.
+        /// </summary>
+        internal static string Parse(this string msg, List<string>? templates)
+        {
+            ReadOnlySpan<char> format = msg.AsSpan();
+            var builder = new StringBuilder(msg.Length);
+            var pos = 0;
+            var len = format.Length;
```

Comment:
>len [](http://example.com/codeflow?start=16&length=3)

i think caching the sapn length will prevent the jit from optimizaing the usage. maybe you can just always call format.Length instead.